### ISSUE TITLE
java.lang.ClassNotFoundException: sun.misc.VM thrown by Docker image on 0.19.x

### ISSUE THREAD

**Issue by rafacm** [gentics/mesh#397]
## Gentics Mesh Version, operating system, or hardware.
Affects:
- v0.19.1
- v0.19.0

Does NOT affect:
- v0.18.3

## Operating System
- macOS Sierra 10.12.6
```
➜  ~ uname -a
Darwin freddie-2.local 16.7.0 Darwin Kernel Version 16.7.0: Tue Jan 30 11:27:06 PST 2018; root:xnu-3789.73.11~1/RELEASE_X86_64 x86_64
```
## Docker Community Edition
```
➜  ~ docker --version
Docker version 18.03.1-ce, build 9ee9f40
```
## Problem
When starting the docker container with
```
➜  ~ docker run -p 8080:8080 gentics/mesh-demo
```
I can see the following warnings exceptions that I did not see in previous 0.18.x versions:
```
13:35:09.455 [] INFO  [main] - Extracting demo data since this is the first time you start mesh...
13:35:12.008 [] INFO  [main] - Demo data extracted to {/mesh/data}
13:35:12.012 [] INFO  [main] - Configuration file {mesh.yml} was not found within classpath.
13:35:12.013 [] INFO  [main] - Configuration file {config/mesh.yml} was not found within filesystem.
13:35:12.375 [] INFO  [main] - Saved default configuration to file {/mesh/config/mesh.yml}.
13:35:12.375 [] INFO  [main] - Loading default configuration.
13:35:12.394 [Breathtaking Trapinch] INFO  [main] - Could not find keystore {config/keystore.jceks}. Creating one for you..
13:35:12.655 [Breathtaking Trapinch] INFO  [main] - Keystore {config/keystore.jceks} created. The keystore password is listed in your {mesh.yml} file.
13:35:12.658 [Breathtaking Trapinch] INFO  [main] - ###############################################################
13:35:12.661 [Breathtaking Trapinch] INFO  [main] - # Mesh Version 0.19.1 2018-04-30T14:19:26Z                    #
13:35:12.661 [Breathtaking Trapinch] INFO  [main] - # Gentics Software                                            #
13:35:12.661 [Breathtaking Trapinch] INFO  [main] - #-------------------------------------------------------------#
13:35:12.666 [Breathtaking Trapinch] INFO  [main] - # Vert.x Version: 3.5.0                                       #
13:35:12.666 [Breathtaking Trapinch] INFO  [main] - # Mesh Node Name: Breathtaking Trapinch                       #
13:35:12.666 [Breathtaking Trapinch] INFO  [main] - ###############################################################
13:35:12.792 [Breathtaking Trapinch] INFO  [main] - Creating orientdb distributed server configuration file {config/default-distributed-db-config.json}
13:35:12.793 [Breathtaking Trapinch] INFO  [main] - Creating orientdb hazelcast configuration file {config/hazelcast.xml}
13:35:12.794 [Breathtaking Trapinch] INFO  [main] - Creating orientdb server configuration file {config/orientdb-server-config.xml}
13:35:12.795 [Breathtaking Trapinch] INFO  [main] - Creating orientdb backup configuration file {config/automatic-backup.json}
13:35:12.796 [Breathtaking Trapinch] INFO  [main] - Creating orientdb server security configuration file {config/security.json}
13:35:13.110 [Breathtaking Trapinch] INFO  [main] - Creating non-clustered Vert.x instance
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by io.netty.util.internal.ReflectionUtil (file:/mesh/meshdemo.jar) to constructor java.nio.DirectByteBuffer(long,int)
WARNING: Please consider reporting this to the maintainers of io.netty.util.internal.ReflectionUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
[...]
WARNING: Unable to determine the amount of MaxDirectMemorySize.
java.lang.ClassNotFoundException: sun.misc.VM
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:582)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:190)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:499)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:291)
	at com.orientechnologies.common.util.OMemory.getConfiguredMaxDirectMemory(OMemory.java:74)
	at com.orientechnologies.orient.core.engine.OMemoryAndLocalPaginatedEnginesInitializer.configureDefaultDiskCacheSize(OMemoryAndLocalPaginatedEnginesInitializer.java:174)
	at com.orientechnologies.orient.core.engine.OMemoryAndLocalPaginatedEnginesInitializer.configureDefaults(OMemoryAndLocalPaginatedEnginesInitializer.java:77)
	at com.orientechnologies.orient.core.engine.OMemoryAndLocalPaginatedEnginesInitializer.initialize(OMemoryAndLocalPaginatedEnginesInitializer.java:62)
	at com.orientechnologies.orient.core.engine.local.OEngineLocalPaginated.startup(OEngineLocalPaginated.java:56)
	at com.orientechnologies.orient.core.Orient.startEngine(Orient.java:930)
	at com.orientechnologies.orient.core.Orient.loadStorage(Orient.java:523)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.<init>(ODatabaseDocumentTx.java:173)
	at com.orientechnologies.orient.core.db.OPartitionedDatabasePool$DatabaseDocumentTxPooled.<init>(OPartitionedDatabasePool.java:428)
	at com.orientechnologies.orient.core.db.OPartitionedDatabasePool$DatabaseDocumentTxPooled.<init>(OPartitionedDatabasePool.java:424)
	at com.orientechnologies.orient.core.db.OPartitionedDatabasePool.initQueue(OPartitionedDatabasePool.java:358)
	at com.orientechnologies.orient.core.db.OPartitionedDatabasePool.<init>(OPartitionedDatabasePool.java:117)
	at com.tinkerpop.blueprints.impls.orient.OrientGraphFactory.<init>(OrientGraphFactory.java:174)
	at com.tinkerpop.blueprints.impls.orient.OrientGraphFactory.<init>(OrientGraphFactory.java:155)
	at com.tinkerpop.blueprints.impls.orient.OrientGraphFactory.<init>(OrientGraphFactory.java:131)
	at com.gentics.mesh.graphdb.OrientDBDatabase.initGraphDB(OrientDBDatabase.java:239)
	at com.gentics.mesh.graphdb.OrientDBDatabase.setupConnectionPool(OrientDBDatabase.java:227)
	at com.gentics.mesh.cli.BootstrapInitializerImpl.init(BootstrapInitializerImpl.java:280)
	at com.gentics.mesh.cli.MeshImpl.run(MeshImpl.java:139)
	at com.gentics.mesh.cli.MeshImpl.run(MeshImpl.java:103)
	at com.gentics.mesh.demo.DemoRunner.main(DemoRunner.java:83)

May 01, 2018 1:35:24 PM com.orientechnologies.common.log.OLogManager log
INFO: OrientDB auto-config DISKCACHE=498MB (heap=500MB direct=500MB os=1,998MB), assuming maximum direct memory size equals to maximum JVM heap size
May 01, 2018 1:35:24 PM com.orientechnologies.common.log.OLogManager log
WARNING: Unable to determine the amount of MaxDirectMemorySize.
java.lang.ClassNotFoundException: sun.misc.VM
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:582)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:190)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:499)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:291)
	at com.orientechnologies.common.util.OMemory.getConfiguredMaxDirectMemory(OMemory.java:74)
	at com.orientechnologies.common.util.OMemory.checkDirectMemoryConfiguration(OMemory.java:103)
	at com.orientechnologies.orient.core.engine.OMemoryAndLocalPaginatedEnginesInitializer.initialize(OMemoryAndLocalPaginatedEnginesInitializer.java:64)
	at com.orientechnologies.orient.core.engine.local.OEngineLocalPaginated.startup(OEngineLocalPaginated.java:56)
	at com.orientechnologies.orient.core.Orient.startEngine(Orient.java:930)
	at com.orientechnologies.orient.core.Orient.loadStorage(Orient.java:523)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.<init>(ODatabaseDocumentTx.java:173)
	at com.orientechnologies.orient.core.db.OPartitionedDatabasePool$DatabaseDocumentTxPooled.<init>(OPartitionedDatabasePool.java:428)
	at com.orientechnologies.orient.core.db.OPartitionedDatabasePool$DatabaseDocumentTxPooled.<init>(OPartitionedDatabasePool.java:424)
	at com.orientechnologies.orient.core.db.OPartitionedDatabasePool.initQueue(OPartitionedDatabasePool.java:358)
	at com.orientechnologies.orient.core.db.OPartitionedDatabasePool.<init>(OPartitionedDatabasePool.java:117)
	at com.tinkerpop.blueprints.impls.orient.OrientGraphFactory.<init>(OrientGraphFactory.java:174)
	at com.tinkerpop.blueprints.impls.orient.OrientGraphFactory.<init>(OrientGraphFactory.java:155)
	at com.tinkerpop.blueprints.impls.orient.OrientGraphFactory.<init>(OrientGraphFactory.java:131)
	at com.gentics.mesh.graphdb.OrientDBDatabase.initGraphDB(OrientDBDatabase.java:239)
	at com.gentics.mesh.graphdb.OrientDBDatabase.setupConnectionPool(OrientDBDatabase.java:227)
	at com.gentics.mesh.cli.BootstrapInitializerImpl.init(BootstrapInitializerImpl.java:280)
	at com.gentics.mesh.cli.MeshImpl.run(MeshImpl.java:139)
	at com.gentics.mesh.cli.MeshImpl.run(MeshImpl.java:103)
	at com.gentics.mesh.demo.DemoRunner.main(DemoRunner.java:83)
```
Attached full log.

## Expected behaviour
Starting the docker container should not throw exceptions.
See file [gentics-mesh-version-0-18-3-docker-java-sun-misc-vm-exception-log.txt](https://github.com/gentics/mesh/files/1966167/gentics-mesh-version-0-18-3-docker-java-sun-misc-vm-exception-log.txt)

## Actual behaviour
Logs show warnings and exceptions.
See files:
1. [gentics-mesh-version-0-19-0-docker-java-sun-misc-vm-exception-log.txt](https://github.com/gentics/mesh/files/1966168/gentics-mesh-version-0-19-0-docker-java-sun-misc-vm-exception-log.txt)
1. [gentics-mesh-version-0-19-1-docker-java-sun-misc-vm-exception-log.txt](https://github.com/gentics/mesh/files/1966169/gentics-mesh-version-0-19-1-docker-java-sun-misc-vm-exception-log.txt)

**Comment by Jotschi** [comment]
Related to internal issue - https://www.prjhub.com/#/issues/10115 (Java 10 Warnings)

**Comment by Jotschi** [comment]
We reverted the update to JRE 10 for now. We'll revisit this in the future.
Fix has been released with 0.19.2
